lib/src/Base/Stat/CovarianceModelImplementation.cxx

```diff
@@ -791,9 +791,63 @@ Bool CovarianceModelImplementation::isDiagonal() const
 /* Marginal accessor */
 CovarianceModel CovarianceModelImplementation::getMarginal(const UnsignedInteger index) const
 {
-  if (index >= outputDimension_) throw InvalidArgumentException(HERE) << "Error: index=" << index << " must be less than output dimension=" << outputDimension_;
-  if (outputDimension_ != 1) throw NotYetImplementedException(HERE) << "In CovarianceModelImplementation::getMarginal(const UnsignedInteger index) const";
-  return clone();
+  return getMarginal(Indices(1, index));
+}
+
+/* Marginal accessor */
+CovarianceModel CovarianceModelImplementation::getMarginal(const Indices & indices) const
+{
+  if (!indices.check(outputDimension_)) throw InvalidArgumentException(HERE) << "Error: index=" << indices << " must be less than output dimension=" << outputDimension_;
+  CovarianceModel result(clone());
+  if (outputDimension_ > 1)
+  {
+    const UnsignedInteger oldOutputDimension = outputDimension_;
+    result.getImplementation()->outputDimension_ = indices.getSize();
+    Point newAmplitude;
+    for (UnsignedInteger i = 0; i < indices.getSize(); ++ i)
+    {
+      newAmplitude.add(amplitude_[indices[i]]);
+    }
+    result.getImplementation()->amplitude_ = newAmplitude;
```

Comment:
I think we should use setter. Some models have to calibrate their internal parameters

Reply:
ok